cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -125,19 +130,24 @@ protected int getDiffSize() {
         return diff.size();
     }
 
-    private static boolean isIgnorableProperty(String key) {
-        return IGNORABLE_PROPERTIES.matcher(key).matches();
+    private static boolean isIgnorableProperty(String key, boolean nodeIsController) {
+        // If node is not a KRaft controller, ignore KRaft controller config properties.
+        if (!nodeIsController) {
+            return IGNORABLE_PROPERTIES.matcher(key).matches() || IGNORABLE_CONTROLLER_PROPERTIES.matcher(key).matches();
+        } else {
+            return IGNORABLE_PROPERTIES.matcher(key).matches();
+        }
     }
 
     /**
      * Computes diff between two maps. Entries in IGNORABLE_PROPERTIES are skipped
-     * @param brokerId id of compared broker
+     * @param nodeRef node reference of compared broker
      * @param desired desired configuration, may be null if the related ConfigMap does not exist yet or no changes are required
      * @param brokerConfigs current configuration
      * @param configModel default configuration for {@code kafkaVersion} of broker
      * @return Collection of AlterConfigOp containing all entries which were changed from current in desired configuration
      */
-    private Collection<AlterConfigOp> diff(int brokerId, String desired,
+    private Collection<AlterConfigOp> diff(NodeRef nodeRef, String desired,
```

Comment:
I would agree this Tom. This class is now more likely something related to nodes in general (with KRaft coming) so even the class name could be `KafkaNodeConfigurationDiff`.

Reply:
Wait, this method is not just for brokers but also for controllers, or? I think everywhere in the codebase we decided to refer to those as nodes. I.e. we have brokers, controllers -> and all of them are nodes.

Reply:
This method is only used for brokers (or combined), not for pure controllers. Everywhere else we do refer to them as `nodes` but since this particular class is only used for nodes that have broker role, so we could make that more explicit I guess. 

I would prefer to use `brokerNodeRef` instead of `brokerRef` since we are talking about nodeRef for a broker.

Reply:
If in the future, we are going to use the same class to do the configuration diff for controller nodes, I suppose we could change them to `nodes` instead as well as changing the classname to KafkaNodeConfigurationDiff as @ppatierno suggested.

Reply:
> This method is only used for brokers (or combined), not for pure controllers.

I had a quick look at where KafkaRoller is using this class, I don't see any condition excluding the usage of this class if the node is a controller. For example in the  `checkIfRestartOrReconfigureRequired` called by `restartIfNecessary` called by `schedule` which can receive a controller node as well when called by `rollingRestart`. Maybe is there anything I am missing?

Reply:
In `checkIfRestartOrReconfigureRequired`, it will initialise `KafkaBrokerConfigurationDiff` if `nodeRef.broker()` is true.

Reply:
You are right, the `if` condition is way above the initialization, I missed that, thanks.

Reply:
no problem :)